tracer/src/Datadog.Trace/Vendors/MessagePack/LZ4/Codec/LZ4Codec.Helper.cs

```diff
@@ -8,7 +8,7 @@ namespace Datadog.Trace.Vendors.MessagePack.LZ4
 {
     internal static partial class LZ4Codec
     {
-#if NETSTANDARD || NETFRAMEWORK
+#if ENABLE_UNSAFE_MSGPACK
```

Comment:
This seems to be a bug in the original code because the method `LZ4Codec.Encode32Unsafe()` and `LZ4Codec.Encode64Unsafe()` are themselves declared behind that.

Reply:
It appears `ENABLE_UNSAFE_MSGPACK` is _always_ set, in the csproj: https://github.com/neuecc/MessagePack-CSharp/blob/v1.9.11/src/MessagePack/MessagePack.csproj#L7

Reply:
Do we/should we set `ENABLE_UNSAFE_MSGPACK` (or remove it entirely)? It appears it's only on unity that it doesn't work (according to the comments) and we don't support that anyway?

Reply:
Good point. We don't set `ENABLE_UNSAFE_MSGPACK ` yes, but I think we can now and enable all the "unsafe" MessagePack code since we added ` <AllowUnsafeBlocks>true</AllowUnsafeBlocks>` a few days ago (#2293). We may even get a small performance boost.

Are there any restriction to using unsafe code? Will it prevent users from running the tracer in some restricted environments?

Reply:
No I don't _think_ there's any restrictions like with security permissions. "Unsafe" should just indicate where the compiler can no longer guarantee that you've written verifiably safe code. If someone can back me up and say "There are _definitely_ no restrictions" would make me feel better 😅

Reference: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/unsafe-code

Reply:
I can't find any official docs on this, but it seems like assemblies with unsafe code cannot be loaded in partial trust scenarios, so we may need to look into #2293 before the next release.

Reply:
Thanks for the comments. I'm gonna merge as-is for now and we can come back to this later.

Reply:
#2331 reverted the PR that enabled unsafe code, so we want' enabled `ENABLE_UNSAFE_MSGPACK`, either. Wo can come back to this if we ever enable unsafe code. Resolving this comment.